app/components/Map.tsx

```diff
@@ -1,20 +1,46 @@
-'use client';
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { MapContainer, TileLayer } from 'react-leaflet';
 import 'leaflet/dist/leaflet.css';
 
 import Antennas from './Antennas';
 import SectorLobes from './SectorLobes';
 import AntennaInfo from './AntennaInfo';
 
-import { AccessPoint } from '../types';
+import { AccessPoint, Antenna } from '../types';
 
 export default function Map() {
   const [toggleInfo, setToggleInfo] = useState(false);
-
   const [currentAntenna, setCurrentAntenna] = useState<AccessPoint | null>(
     null
   );
+  const [antennasData, setAntennasData] = useState<AccessPoint[]>([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch('./app/api/v1/antenna/');
+        if (!response.ok) {
+          throw new Error(`${response.status} error: Failed to fetch atnennas`);
+        }
+        const accessPoints = (await response.json()) as Antenna[];
+
+        const antennasData: AccessPoint[] = accessPoints.map((ap: Antenna) => ({
+          id: ap.id,
+          modelName: ap.modelname,
+          lat: ap.latitude,
+          lon: ap.longitude,
+          initialHeading: ap.initialheading,
+          heading: ap.heading,
+          radius: ap.radius,
+        }));
+
+        setAntennasData(antennasData);
+      } catch (e) {
+        console.log(e);
```

Comment:
@JonMike8 Instead of just console logging the error, we can follow best practices not to affect UX.
1. Add a retry logic
2. Display alert

## Notes of implementation of Retry logic
You could try something like
```
async function fetchDataWithRetry(url, maxRetries = 3, retryDelay = 1000) {
    for (let attempt = 1; attempt <= maxRetries; attempt++) {
        try {
            const response = await fetch(url);
            if (!response.ok) {
                throw new Error(`${response.status} error: Failed to fetch`);
            }
            return await response.json(); // Assuming the response is JSON
        } catch (e) {
            console.error(`Attempt ${attempt} failed: ${e.message}`);
            if (attempt === maxRetries) throw e; // Rethrow error on last attempt
            await new Promise(resolve => setTimeout(resolve, retryDelay)); // Wait before retrying
        }
    }
}

const URL = '/api/v1/antenna/';
fetchDataWithRetry(URL)
    .then(data => {
    ...
    })
    .catch(error => {
        console.error('error');
    });
```

## Notes on alert display
```
catch (e) {
  switch(e.statusCode) {
    case 404:
      displayErrorMessage("The requested resource was not found.");
      break;
    case 500:
      displayErrorMessage("A server error occurred. Please try again later.");
      break;
    default:
      displayErrorMessage("An unexpected error occurred.");
  }
}
```
You have to implement ```displayErrorMessage``` which just displays an alert box to the user displaying the error message.